build: make libxml2 handling explicit and use in tests

compnerd opened the pull request:
Address the TODO in the build and unify the libxml2 handling with
LLVM/clang.  Use the information now and propagate this into the lit
configuration so that tests can be made aware of libxml2 status.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

gottesmm commented on test/CMakeLists.txt:
```diff
@@ -14,6 +14,12 @@ function(swift_configure_lit_site_cfg source_path destination_path installed_nam
     set(LIT_SWIFTLIB_DIR ${SWIFTLIB_DIR})
   endif ()
 
+  if(SWIFT_HAVE_LIBXML)
```
Can you use a different variable name here for the configure? I am nervous about people getting confused. Changing the name would make me more confortable.

compnerd replied:
Ugh, naming is hard.  I think that an alternate name is a great idea, but I don't really have any good suggestions.  Mind providing some suggestions?

gottesmm replied:
LIT_SWIFT_HAVE_LIBXML

compnerd replied:
BTW, `lit.site.cfg` has a number of the variables with the same value.  I think that the problem is that the value being saved here is not guaranteed to be normalized :-(.

compnerd commented on CMakeLists.txt:
```diff
@@ -867,12 +867,13 @@ find_package(PythonInterp REQUIRED)
+if(LLVM_ENABLE_LIBXML2)
+  set(LIBXML2_FOUND NO)
+  find_package(LibXml2)
+  if(LIBXML2_FOUND)
```
The version request only makes sense if there is a dependency on an API that was introduced after the initial release.  I don't know if that is the case.  I don't see any harm in requesting the minimum version to be the same as what is shipped on macOS just to ensure that we have the same base line of support.  The problem is, I don't know what the version on macOS is :-p.

I don't really see there being a benefit to the 2 lines being in LLVM.

gottesmm replied:
Ok. I just wasn't sure what the comment was asking for (in terms of the 2 lines).

@jrose-apple do you know the version of libxml we should require here?

compnerd replied:
Ping!  I want to get this handled as I'm trying to handle the last of the failures on Windows.

jrose-apple replied:
Oops, I must have missed this the first time around. We use libxml for doc comment processing, so @nkcsgexi would probably know what our dependencies are.

compnerd replied:
@jrose-apple, yeah, which has some tests which should be conditional on whether we built with libxml2 or not.

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@jrose-apple - any thoughts on whether LibXml2 should be required on Linux or make it optional?

compnerd commented:
@swift-ci please test

jrose-apple commented:
Again, that's a question for the SourceKit / source tooling folks rather than me.

compnerd committed:
build: make libxml2 handling explicit and use in tests

Address the TODO in the build and unify the libxml2 handling with
LLVM/clang. Use the information now and propagate this into the lit
configuration so that tests can be made aware of libxml2 status.

compnerd commented:
@swift-ci please smoke test

compnerd commented:
I'm going to merge this since this is needed for the Windows test suite.  This is no worse than the status quo, and we can add the versioning in a follow up.  This effectively makes `-DLLVM_ENABLE_LIBXML2` a knob the user can use to ensure that libxml2 support is present, otherwise it falls back to the auto-detect as it was previously.
